cyberbrain/value_stack.py

```diff
@@ -137,6 +217,7 @@ def emit_event_and_update_stack(
         """
         self.snapshot = snapshot
         opname = instr.opname
+        _placeholder.start_lineno = self.last_starts_line = lineno
```

Comment:
Can you think of a way to not requiring modifying `_placeholder`? It's a global value so we shouldn't modify it. Plus, I don't see where `_placeholder.start_lineno` is used.

Reply:
When you LOAD_CONST and then STORE_VALUE, you push a `_placeholder` onto the stack and then create an event off of that placeholder, taking its `start_lineno`.  Therefore, the lineno needs to either exist in the stack or every event creation needs to check if lineno is -1 and act accordingly (which may also cause issues if LOAD and STORE are on different lines). We could change `_placeholder` to be a function that takes in a lineno and returns the corresponding SymbolStackItem instead or add a check in `_push` (if _placeholder: set lineno).

Reply:
I see. I'm concerned with the current implementation because in the long term cb will support tracing multiple frames, and if they share a mutable _placeholder, things will not work. Now it looks like every frame should have its own placeholder. And since every frame has a value stack, an option is to move it to be an instance variable. wdyt?

Reply:
In the most up to date version, `_placeholder` is only used 8 times, so maybe it'd be better to just use the full form (`SymbolStackItem(self.last_starts_line, [])`)? Otherwise yeah it makes sense to create a ValueStack attribute since it uses the ValueStack's `last_starts_line` attribute anyway and would support multi-frame tracing.

Reply:
For simplicity, moving it inside the class feels like a better approach to me

Reply:
Done